[BUG] SQLite db file not found after upgrade to 3.9.5

After performing meshmonitor upgrade from 3.9.4 to 3.9.5, meshmonitor will not initialize due to SQLite being unable to find the database file.

The steps I took today were the same as before: See the new release banner, SSH into the Debian VM running docker, execute a docker pull, down, up -d in that order. Logs indicate the database file is not found. I can confirm the file is still there, and the docker volume config did not change.

docker-compose:
```
services:
  ble-bridge:
    image: ghcr.io/yeraze/meshtastic-ble-bridge:latest
    container_name: meshtastic-ble-bridge
    restart: unless-stopped
    privileged: true
    volumes:
      - /var/run/dbus:/var/run/dbus
      - /var/lib/bluetooth:/var/lib/bluetooth:ro
    command: --port 4403 C9:0A:F8:82:36:25

  meshmonitor:
    image: ghcr.io/yeraze/meshmonitor:latest
    volumes:
      - /srv/meshmonitor/data:/data
    container_name: meshmonitor
    restart: unless-stopped
    ports:
      - "8080:3001"
      - "4404:4404"
    environment:
      - TZ=America/New_York
      - COOKIE_SECURE=true
      - ALLOWED_ORIGINS=http://192.168.10.83:8080,https://meshmonitor.izzyhizzy.us
      - MESHTASTIC_NODE_IP=ble-bridge
      - MESHTASTIC_TCP_PORT=4403
      - ENABLE_VIRTUAL_NODE=true
      - VIRTUAL_NODE_POST=4404

  mqtt-proxy:
    image: ghcr.io/ln4cy/mqtt-proxy:master
    container_name: meshmonitor-mqtt-proxy
    restart: unless-stopped
    environment:
      - INTERFACE_TYPE=tcp
      - TCP_NODE_HOST=meshmonitor
      - TCP_NODE_PORT=4404
      - LOG_LEVEL=INFO
      - TCP_TIMEOUT=300
      - CONFIG_WAIT_TIMEOUT=60
      - HEALTH_CHECK_ACTIVITY_TIMEOUT=300
    depends_on:
      - meshmonitor
    healthcheck:
      test: ["CMD-SHELL", "test -f /tmp/healthy && find /tmp/healthy -mmin -1 | grep -q healthy"]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 60s
```

startup logs:
```
2026-03-16 20:12:45,267 INFO success: meshmonitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
[WARN] 
[WARN] ═══════════════════════════════════════════════════════════
[WARN] ⚠️  SESSION_SECRET NOT SET - USING AUTO-GENERATED SECRET
[WARN] ═══════════════════════════════════════════════════════════
[WARN]    For basic/home use, this is OK. Sessions will work.
[WARN]
[WARN]    For production deployments with HTTPS, set SESSION_SECRET:
[WARN]    SESSION_SECRET=$(openssl rand -hex 32)
[WARN]
[WARN]    ⚠️  Sessions will be reset on each container restart!
[WARN] ═══════════════════════════════════════════════════════════
[WARN]
[INFO] 📋 Environment configuration:
[INFO]    NODE_ENV: production (env)
[INFO]    PORT: 3001 (env)
[INFO]    BASE_URL: / (default)
[INFO]    LOG_LEVEL: info (default)
[INFO]    TZ: America/New_York (env)
[INFO]    ALLOWED_ORIGINS: http://192.168.10.83:8080,https://meshmonitor.izzyhizzy.us (env)
[INFO]    TRUST_PROXY: false (default)
[INFO]    VERSION_CHECK_DISABLED: false
[INFO]    --- Session/Security ---
[INFO]    SESSION_SECRET: (auto-generated)
[INFO]    SESSION_COOKIE_NAME: meshmonitor.sid (default)
[INFO]    SESSION_MAX_AGE: 86400000ms (default)
[INFO]    SESSION_ROLLING: true (default)
[INFO]    COOKIE_SECURE: true (env)
[INFO]    COOKIE_SAMESITE: lax (default)
[INFO]    --- Database ---
[INFO]    DATABASE_TYPE: sqlite
[INFO]    DATABASE_PATH: /data/meshmonitor.db (default)
[INFO]    --- Meshtastic ---
[INFO]    MESHTASTIC_NODE_IP: ble-bridge (env)
[INFO]    MESHTASTIC_TCP_PORT: 4403 (env)
[INFO]    MESHTASTIC_STALE_CONNECTION_TIMEOUT: 300000ms (default)
[INFO]    MESHTASTIC_CONNECT_TIMEOUT_MS: 10000ms (default)
[INFO]    MESHTASTIC_RECONNECT_INITIAL_DELAY_MS: 1000ms (default)
[INFO]    MESHTASTIC_RECONNECT_MAX_DELAY_MS: 60000ms (default)
[INFO]    MESHTASTIC_MODULE_CONFIG_DELAY_MS: 100ms (default)
[INFO]    --- Virtual Node ---
[INFO]    ENABLE_VIRTUAL_NODE: true (env)
[INFO]    VIRTUAL_NODE_PORT: 4404 (default)
[INFO]    VIRTUAL_NODE_ALLOW_ADMIN_COMMANDS: false (default)
[INFO]    --- Authentication ---
[INFO]    DISABLE_ANONYMOUS: false (default)
[INFO]    DISABLE_LOCAL_AUTH: false (default)
[INFO]    --- Rate Limiting ---
[INFO]    RATE_LIMIT_API: 1000 req/min (default)
[INFO]    RATE_LIMIT_AUTH: 5 req/min (default)
[INFO]    RATE_LIMIT_MESSAGES: 30 req/min (default)
[ERROR] ❌ DATABASE OPEN ERROR ❌
[ERROR] ═══════════════════════════════════════════════════════════
[ERROR] Failed to open SQLite database at: /data/meshmonitor.db
[ERROR]
[ERROR] Error: disk I/O error
[ERROR] Error code: SQLITE_IOERR_SHMSIZE
[ERROR] ═══════════════════════════════════════════════════════════
file:///app/dist/services/database.js:355
            throw new Error(`Database initialization failed: ${err.message}`);
                  ^

Error: Database initialization failed: disk I/O error
    at new DatabaseService (file:///app/dist/services/database.js:355:19)
    at file:///app/dist/services/database.js:11759:16
    at ModuleJob.run (node:internal/modules/esm/module_job:430:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:661:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:101:5)

Node.js v24.14.0
npm notice
npm notice New minor version of npm available! 11.9.0 -> 11.11.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.11.1
npm notice To update run: npm install -g npm@11.11.1
npm notice
^C2026-03-16 20:12:47,403 WARN exited: meshmonitor (exit status 1; not expected)
2026-03-16 20:12:48,409 INFO spawned: 'meshmonitor' with pid 250
```

--- a/src/services/database.ts
+++ b/src/services/database.ts
@@ -621,13 +621,7 @@ class DatabaseService {
     }
 
     // Now attempt to open the database with better error handling
-    try {
-      this.db = this.openSqliteDatabase(dbPath, dbDir);
-    } catch (error: unknown) {
-      const err = error as Error & { code?: string };
-      logger.error('═══════════════════════════════════════════════════════════');
-      throw new Error(`Database initialization failed: ${err.message}`);
-    }
+    this.db = this.openSqliteDatabase(dbPath, dbDir);
 
     // Initialize models
     this.userModel = new UserModel(this.db);
@@ -5204,11 +5198,12 @@ class DatabaseService {
 
       // Stale SHM file from a previous version — remove it and retry
       const shmPath = `${dbPath}-shm`;
-      if (err.code?.startsWith('SQLITE_IOERR') && fs.existsSync(shmPath)) {
+      const isShmError = err.code === 'SQLITE_IOERR_SHMSIZE' || err.code === 'SQLITE_IOERR_SHMMAP';
+      if (isShmError) {
         logger.warn('⚠️  SQLite SHM file appears stale (common after upgrades)');
         logger.warn(`   Removing ${shmPath} and retrying — data is safe in the WAL`);
+        fs.rmSync(shmPath, { force: true });
         try {
-          fs.unlinkSync(shmPath);
           return attemptOpen();
         } catch (retryError: unknown) {
           const retryErr = retryError as Error & { code?: string };

